lib/libc/minimal/source/string/string.c

```diff
@@ -170,6 +170,37 @@ int strncmp(const char *s1, const char *s2, size_t n)
 	return (n == 0) ? 0 : (*s1 - *s2);
 }
 
+/**
+ * @brief Separate `str` by any char in `sep` and return NULL terminated
+ * sections. Consecutive `sep` chars in `str` are treated as a single
+ * separator.
+ *
+ * @return pointer to NULL terminated string or NULL on errors.
+ */
+char *strsep(char **str, const char *sep)
+{
+	char *start, *end;
+
+	if (str == NULL || *str == NULL || *str[0] == '\0') {
+		return NULL;
+	}
+
+	start = *str;
+	while (*start && strchr(sep, *start)) {
+		start++;
+	}
+	end = start;
+	while (*end && !strchr(sep, *end)) {
```

Comment:
This appears to be skipping over a sequence of separators without generating empty tokens, which is inconsistent with standard `strsep`.  See comment in test case.